### PR TITLE
Add config and jobs for openshift/cloud-credential-operator

### PR DESCRIPTION
**Pull request by dgoodwin**
Replaces https://github.com/openshift/release/pull/2445

CC @smarterclayton @petr-muller

### CODE REVIEW

File: ci-operator/config/openshift/cloud-credential-operator/openshift-cloud-credential-operator-master.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      cpu: 100m
+      memory: 200Mi
+tag_specification:
+  cluster: https://api.ci.openshift.org
+  name: origin-v4.0
+  namespace: openshift
+  tag: ''
+  tag_overrides: {}
+build_root:
+  #  image_stream_tag:
```
Comment (smarterclayton) [inline ci-operator/config/openshift/cloud-credential-operator/openshift-cloud-credential-operator-master.yaml]: Why are you building your own base image?

Comment (dgoodwin) [reply ci-operator/config/openshift/cloud-credential-operator/openshift-cloud-credential-operator-master.yaml]: We have a few build time dependencies (gomock, mockgen) and via the docs it looked like this was the correct way to handle that (https://github.com/openshift/ci-operator/blob/master/CONFIGURATION.md#build_root), it's what we did in Hive as well. Our image is based on openshift/origin-release:golang-1.10.

Is this incorrect use of the config and is there a better way?

File: cluster/ci/config/prow/config.yaml [hunk 2]
```diff
@@ -181,6 +181,7 @@ tide:
     - do-not-merge/work-in-progress
     - do-not-merge/invalid-owners-file
   - repos:
+    - openshift/cloud-credential-operator
```
[the remark this answers is not in the document]
Comment (dgoodwin) [reply cluster/ci/config/prow/config.yaml]: Is this approve plugin required? It's not desired on my end, we don't have a team of size that needs it and the default inherited set would be fine.
I've added in any case but if it's ok to remove again I'd prefer to do that.

File: ci-operator/config/openshift/cloud-credential-operator/openshift-cloud-credential-operator-master.yaml (hunk 1 above)
Comment (dgoodwin) [reply ci-operator/config/openshift/cloud-credential-operator/openshift-cloud-credential-operator-master.yaml]: I think I found a solution, the deps in question are for generation which shouldn't be done for a CI build, it should just use the committed code. Added some make targets for CI that skip generation, and I'm now able to use the standard base image.

File: cluster/ci/config/prow/config.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (dgoodwin) [reply cluster/ci/config/prow/config.yaml]: Need more clarification here @smarterclayton, I've removed the approve plugin again for now. Why do we need to be in plugins.yaml?

**Comment by dgoodwin** [comment]
Via @smarterclayton on Slack, everyone should be using /approve unless you're really special and none of us are special.

Re-added to plugins, moved to config list that requires approve and lgtm.

Should be good to go now.

**Pull request merged**